dali/pipeline/workspace/device_workspace.cc

```diff
@@ -36,6 +36,24 @@ const TensorList<GPUBackend>& DeviceWorkspace::Input(int idx) const {
   return *gpu_inputs_[tensor_meta.second];
 }
 
+template <>
+TensorList<CPUBackend>& DeviceWorkspace::MutableInput(int idx) {
+  DALI_ENFORCE_VALID_INDEX(idx, input_index_map_.size());
+  auto tensor_meta = input_index_map_[idx];
+  DALI_ENFORCE(tensor_meta.first, "Input TensorList with given "
+      "index does not have the calling backend type (CPUBackend)");
+  return *cpu_inputs_[tensor_meta.second];
+}
+
+template <>
+TensorList<GPUBackend>& DeviceWorkspace::MutableInput(int idx) {
```

Comment:
Same as above

Reply:
I am thinking that we could easily merge the impls Input/Output of all the workspaces into a generic one

Reply:
done